### PR TITLE
Updated mutational-signatures: PR 1 of 2

### PR DESCRIPTION
**Pull request by sjspielman**
### Purpose/implementation Section

The `mutational-signatures` module needs to be finalized, with the following overall goals. These goals **replace** previous `05` and `06` scripts/notebooks.

**This PR:** 

1. Perform _extraction_ of the 8 CNS Signal/RefSig signatures with two methods, `deconstructSigs` and `sigfit`. This is now performed in `05-fit_cns_signatures.R`. This script saves two files contained the estimated signature exposures, one per methods:
    + `results/fitted_exposures_signal-cns-deconstructSigs.RDS`
    + `results/fitted_exposures_signal-cns-sigfit.RDS`
2. Perform a _brief_ comparison of the results from these two methods in `06-compare_cns_exposures.Rmd`. Comparison shows overall high similarity.

**Next PR:**

3. Update this module with pub-ready visualizations which will live in an `07` notebook.

In the next PR, the overall analysis `README.md` and the specific `mutational-signatures` module `README.md` will be updated.

#### What GitHub issue does your pull request address?

#1220 

#### Which areas should receive a particularly close look?

Overall organization.

#### Is there anything that you want to discuss further?

There is a world in which `05` _just_ uses `deconstructSigs` and proceeds straight to `07`, skipping any (then moot) comparisons which happen in the current `06`. Plus, because the current approach does a quick evaluation of both methods, we don't actually end up saving time by visualizing the faster method. _But_, skipping the comparison seems less desirable for posterity and having fewer "magic" decisions in the pipeline. 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### What types of results are included (e.g., table, figure)?

New result files in `results/`, as well as some quick scatterplots in the `06` notebook. These are not meant for the publication.

#### What is your summary of the results?

`sigfit` and `deconstructSigs` give fairly consistent results, with the latter method being much faster. So, the latter method will be use for subsequent viz.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/mutational-signatures/06-compare_cns_exposures.Rmd [hunk 1]
```diff
@@ -0,0 +1,172 @@
+---
+title: "Quick comparison of CNS Signal exposures estimated from `sigfit` and `deconstructSigs`"
```
Comment (jaclyn-taroni) [inline analyses/mutational-signatures/06-compare_cns_exposures.Rmd:2]: I don't think we have the most up to date version of the HTML file here.

**Review by jaclyn-taroni — COMMENTED** [review]
This mostly seems to be in order and I think we should leave in the comparison! The HTML doesn't seem to be up to date, which we should remedy before approval and merging. I also have a style comment that we should discuss.

File: analyses/mutational-signatures/06-compare_cns_exposures.Rmd [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  tibble::as_tibble()  %>%
+  dplyr::mutate(type = "sigfit") -> sigfit_exposures
+
+
+# deconstructSigs means exposures into a tibble
+lapply(decon_rds, "[[", "weights") -> decon_mean
+do.call(rbind, decon_mean) %>%
+  tibble::rownames_to_column("sample_id") %>%
+  tibble::as_tibble() %>%
+  dplyr::mutate(type = "decon") -> decon_exposures
```
[the remark this answers is not in the document]
Comment (sjspielman) [reply analyses/mutational-signatures/06-compare_cns_exposures.Rmd:58]: Ok, can fix as part of re-rendering the html

**Review by jaclyn-taroni — APPROVED** [review]
👍 thanks for the changes

**Pull request merged**